platform/ios/scripts/release-notes.js

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const ejs = require('ejs');
+const _ = require('lodash');
+const semver = require('semver');
+
+const changelog = fs.readFileSync('platform/ios/CHANGELOG.md', 'utf8');
+
+// This regular expression parses the changelog for individual releases:
+//  - Matches lines starting with "## ".
+//  - Groups the version number.
+//  - Skips the (optional) release date.
+//  - Groups the changelog content.
+//  - Ends when another "## " is found.
+const regex = /^## (\d+\.\d+\.\d+).*?\n(.+?)(?=\n^## )/gms;
+
+let releases = [];
+while (match = regex.exec(changelog)) {
```

Comment:
Does this means it will parse the whole file each time? If so will it be more performance aware to parse the last two versions e.g. `4.x` and `3.x` or maybe stop at top 10?

Reply:
Feel free to dismiss my comment if the time added to the process is minimum.

Reply:
> Does this means it will parse the whole file each time?

The [`RegExp.exec` docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) say that successive runs start from the previous offset. (I also borrowed the `while` structure from their example.)

This script currently finishes instantaneously (on my admittedly higher-spec’d computer). There are certainly opportunities for optimization (not parsing unnecessary versions, using a less greedy regexp, etc.), but my initial attempt here prioritized readability and minimum developer effort. :wink:

I see this script expanding to handle #11443 (and pre-releases, perhaps), so there will be refactors and we can fix performance issues as they arise.

Reply:
Thank you for explaining. You are right let's worry about performance once we hit an issue.